### DISCUSSION TITLE
Go to input box and highlight text?

### DISCUSSION THREAD

**Discussion by kickerofelves** [tridactyl/tridactyl#5014] [Q&A] [answered]
I want something like ```gi``` but where it also highlights any existing text.  Just like with url bar of the browser.  Is it possible with tridactyl?  Something else if tridactyl can't?

**Comment by Chic-Tweetz** [comment, marked as answer]
This defines an alias `focusinput_highlight` which you can bind to whatever you like:

`alias focusinput_highlight js -dΩ tri.excmds.focusinput(JS_ARGS[1]);let ae=document.activeElement;if (ae.tagName==="INPUT"||ae.tagName==="TEXTAREA")ae.select()Ω`

I used `js` instead of `composite` so `focusinput` can take its usual args.

`gi` is bound to `focusinput -l`, so you'd want to use `bind [key(s)] focusinput_highlight -l` to match its behaviour. 

Incidentally, this was just a small change to this alias which I already had bound to `I` which clears text from input boxes when they're selected.

`alias focusinput_clear js -dΩ tri.excmds.focusinput(JS_ARGS[1]);let ae=document.activeElement;if (ae.tagName==="INPUT"||ae.tagName==="TEXTAREA")ae.value=""Ω`

**Reply by kickerofelves** [reply]
@Chic-Tweetz Great, that works! Thanks!
